### ISSUE TITLE
Support client-side job ID generation with optional user-supplied prefix

### ISSUE THREAD

**Issue by quartzmo** [GoogleCloudPlatform/google-cloud-ruby#1624]
See [Generating a job ID](https://cloud.google.com/bigquery/docs/managing-jobs#generate-jobid) for background.

**Comment by quartzmo** [comment]
The [google-cloud-dotnet implementation](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.JobCrud.cs#L245-L255) uses UUID.

**Comment by blowmage** [comment]
Reading through the docs, I would have a prefix of "job" and use a UUID.

```ruby
prefix = "job"
uuid = SecureRandom.uuid
job_id = "#{prefix}_#{uuid}"
```

So something like this:

```ruby
def query_job query, prefix: "job"
  job_id = "#{prefix}_#{SecureRandom.uuid}"
  # ...
end
```

**Comment by quartzmo** [comment]
I wonder if `SecureRandom#urlsafe_base64` is a better choice than `SecureRandom#uuid`?(@blowmage I think you suggested this first.) It seems closer to the BigQuery service's ids, if that is desirable.

### Examples

BigQuery service:

```
job_12vEDtMQ0mbp1Mo5Z7mzAFQJZazh
```

Using `SecureRandom.uuid`:

```
job_56d91226-205b-4bcc-9bc0-f32a2993f61e
```

Using `SecureRandom.urlsafe_base64(21)`:

```
job_gdQ2fa95Ncbe-2wxdPwlAfX-z6YN
```

**Comment by blowmage** [comment]
I'm definitely in favor of using `SecureRandom.urlsafe_base64`. I might even go so far as to suggest using `SecureRandom.urlsafe_base64(32)`...

**Comment by quartzmo** [comment]
Closed by #1664.
